mock-rollout/src/store/HardwareContext.tsx

```diff
@@ -2,37 +2,41 @@ import React, { createContext, FC, useEffect, useState, useReducer } from 'react
 import { IHardware } from '../model/HardwareModel';
 import { HardwareService } from '../services/HardwareService';
 
-interface IState {
-    hardwares?: IHardware[] | null | undefined;
+interface IHardwareContextState {
```

Comment:
your context state was trying to define both of API accessing methods and the dispatch method to update the context

that's why you differentiate between what is the ContextState and the Context